clients/fides-js/src/lib/consent-utils.ts

```diff
@@ -197,22 +197,40 @@ const hasCurrentPreference = (
   return records.some((record) => record.current_preference);
 };
 
+const hasActionNeededTcfPreference = (
+  records: Pick<TCFPurposeRecord, "current_preference">[] | undefined
+) => {
+  if (!records || records.length === 0) {
+    return false;
+  }
+  return records.some((record) => record.current_preference == null);
+};
+
 /**
  * Returns true if the user has any saved TCF preferences
  */
-export const hasSavedTcfPreferences = (experience: PrivacyExperience) =>
+export const hasNoSavedTcfPreferences = (experience: PrivacyExperience) =>
```

Comment:
why is this named `hasNoSavedTcfPreferences` when it returns true if the user _has_ saved preferences?

Reply:
ack good catch, I kept getting my booleans messed up trying to figure this logic out 🤦‍♀️